### PR TITLE
Initial SYCL backend

### PR DESCRIPTION
**Pull request by masterleinad**
This is meant as a stepping stone to integrate #3231 just providing the most basic functionalities to define the backend and pass the `Serial` tests with `Kokkos_ENABLE_SYCL`. It turns out that this is enough to go up to feature level 2 in the incremental tests.

### CODE REVIEW

**Comment by masterleinad** [comment]
This seems to work now.

File: core/src/impl/Kokkos_Core.cpp [hunk 1]
```diff
@@ -459,6 +481,10 @@ void fence_internal() {
   Kokkos::Experimental::HIP::impl_static_fence();
 #endif
 
+#if defined(KOKKOS_ENABLE_SYCL)
+  Kokkos::Experimental::SYCL().fence();
```
Comment (dalg24) [inline core/src/impl/Kokkos_Core.cpp:485]: No `impl_static_fence()`?

File: core/src/impl/Kokkos_Core.cpp [hunk 2]
```diff
@@ -1219,6 +1257,11 @@ void print_configuration(std::ostream& out, const bool detail) {
 #ifdef KOKKOS_ENABLE_HIP
   Experimental::HIP::print_configuration(msg, detail);
 #endif
+#ifdef KOKKOS_ENABLE_SYCL
+  // FIXME_SYCL
+  std::abort();
+  // Experimental::SYCL::print_configuration(msg, detail);
```
Comment (dalg24) [inline core/src/impl/Kokkos_Core.cpp:1263]: Why is is commented out?  Not implemented?

File: core/src/SYCL/Kokkos_SYCL.cpp [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+void SYCL::impl_finalize() { Impl::SYCLInternal::singleton().finalize(); }
+
+void SYCL::fence() const { m_space_instance->m_queue->wait(); }
+
+SYCL::SYCLDevice::SYCLDevice(cl::sycl::device d) : m_device(std::move(d)) {}
+
+SYCL::SYCLDevice::SYCLDevice(const cl::sycl::device_selector& selector)
+    : m_device(selector.select_device()) {}
+
+cl::sycl::device SYCL::SYCLDevice::get_device() const { return m_device; }
```
[the remark this answers is not in the document]
Comment (dalg24) [reply core/src/SYCL/Kokkos_SYCL.cpp:95]: My bad :)

**Comment by masterleinad** [comment]
Retest this please.

File: core/src/SYCL/Kokkos_SYCL_Instance.cpp [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  if (is_initialized()) return;
+
+  int constexpr WordSize = sizeof(size_type);
+
+  if (!HostSpace::execution_space::impl_is_initialized()) {
+    const std::string msg(
+        "SYCL::initialize ERROR : HostSpace::execution_space is not "
+        "initialized");
+    Kokkos::Impl::throw_runtime_exception(msg);
```
[the remark this answers is not in the document]
Comment (masterleinad) [reply core/src/SYCL/Kokkos_SYCL_Instance.cpp:96]: Just for clarification, you would prefer to either abort in both cases or throw in both cases for all backends?

**Comment by masterleinad** [comment]
Retest this please.

**Comment by masterleinad** [comment]
Closing in favor of #3447.

**Pull request closed without merging**